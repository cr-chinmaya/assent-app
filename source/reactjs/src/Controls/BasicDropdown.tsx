import * as React from 'react';
import { Dropdown } from '@fluentui/react/lib/Dropdown';

const BasicDropdown = ({
    options = [],
    placeholder = '',
    selectedKey = '',
    label = '',
    onChange = (): void => {},
    required = false,
    styles = {},
    componentRef = null,
    errorMessage = null,
    ariaLabel = null,
}: {
    options: any;
    placeholder?: string;
    selectedKey: string;
    label: string;
    onChange: any;
    required?: boolean;
    styles?: object;
    componentRef?: any;
    errorMessage?: any;
    ariaLabel?: string;
}): any => {
    return (
        <Dropdown
            options={options}
            placeholder={placeholder}
            selectedKey={selectedKey}
            label={label}
            onChange={onChange}
            required={required}
            styles={styles}
            componentRef={componentRef}
            errorMessage={errorMessage}
            ariaLabel={ariaLabel}
        />
    );
};
export default BasicDropdown;
