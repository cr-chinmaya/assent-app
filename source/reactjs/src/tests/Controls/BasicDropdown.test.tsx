import * as React from 'react';
import BasicDropdown from '../../Controls/BasicDropdown';
import { render, fireEvent, screen, getByLabelText } from '@testing-library/react';

describe('BasicDropdown', () => {
    it('should BasicDropdown renders correctly', () => {
        const wrapper = render(<BasicDropdown />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should BasicDropdown renders correctly with  props', () => {
        const options = [
            { value: 'red', label: 'Red' },
            { value: 'green', label: 'Green' }
        ];
        const wrapper = render(
            <BasicDropdown
                options={options}
                placeholder={''}
                selectedKey={''}
                label={'Test'}
                required={true}
                styles={() => {}}
            />
        );
        expect(wrapper.getByLabelText('Test')).toBeTruthy;
    });

    it('should BasicDropdown onChange Event fire correctly', () => {
        const text = 'test';
        const options = [
            { value: 'red', label: 'Red' },
            { value: 'green', label: 'Green' }
        ];
        const {getByLabelText} = render(
            <BasicDropdown
                options={options}
                placeholder={''}
                selectedKey={'red'}
                label={text}
                required={true}
                styles={() => {}}
            />
        );
        const dropDown = getByLabelText(text) as HTMLInputElement;
        expect(dropDown).toBeTruthy;
        fireEvent.change(dropDown);
    });
});
