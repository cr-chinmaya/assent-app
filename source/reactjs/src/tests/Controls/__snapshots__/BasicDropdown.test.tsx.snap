// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicDropdown should BasicDropdown renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-Dropdown-container"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-required="false"
          class="ms-Dropdown dropdown-40"
          data-is-focusable="true"
          id="Dropdown0"
          role="listbox"
          tabindex="0"
        >
          <span
            aria-atomic="true"
            aria-invalid="false"
            aria-live="polite"
            aria-setsize="0"
            class="ms-Dropdown-title ms-Dropdown-titleIsPlaceHolder title-41"
            id="Dropdown0-option"
            role="option"
          />
          <span
            class="ms-Dropdown-caretDownWrapper caretDownWrapper-42"
          >
            <i
              aria-hidden="true"
              class="ms-Dropdown-caretDown caretDown-56"
              data-icon-name="ChevronDown"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-Dropdown-container"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-required="false"
        class="ms-Dropdown dropdown-40"
        data-is-focusable="true"
        id="Dropdown0"
        role="listbox"
        tabindex="0"
      >
        <span
          aria-atomic="true"
          aria-invalid="false"
          aria-live="polite"
          aria-setsize="0"
          class="ms-Dropdown-title ms-Dropdown-titleIsPlaceHolder title-41"
          id="Dropdown0-option"
          role="option"
        />
        <span
          class="ms-Dropdown-caretDownWrapper caretDownWrapper-42"
        >
          <i
            aria-hidden="true"
            class="ms-Dropdown-caretDown caretDown-56"
            data-icon-name="ChevronDown"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
